modules/extension/wmts/src/test/java/org/geotools/data/wmts/client/WMTSTransparentTileTest.java

```diff
@@ -0,0 +1,125 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+/**
+ * @author peck7 on 21/12/2017.
+ */
+public class WMTSTransparentTileTest {
+
+    WMTSTileService service;
+
+    @BeforeClass
+    public static void init() {
+        Logger LOGGER = org.geotools.util.logging.Logging.getLogger("org.geotools");
+        LOGGER.setLevel(Level.FINE);
+    }
+
+    @Before
+    public void setup() throws Exception {
+        service = createKVPService();
+    }
+
+    private WMTSTileService createKVPService() throws Exception {
+        try {
+            URL capaResource = getClass().getClassLoader()
+                    .getResource("test-data/geosolutions_getcapa_kvp.xml");
+            assertNotNull("Can't find KVP getCapa file", capaResource);
+            File capaFile = new File(capaResource.toURI());
+            WMTSCapabilities capa = createCapabilities(capaFile);
+
+            String baseURL = "http://demo.geo-solutions.it/geoserver/gwc/service/wmts";
+
+            WMTSLayer layer = capa.getLayer("unesco:Unesco_point");
+            TileMatrixSet matrixSet = capa.getMatrixSet("EPSG:900913");
+            assertNotNull(layer);
+            assertNotNull(matrixSet);
+
+            return new WMTSTileService(baseURL, WMTSServiceType.KVP, layer, null, matrixSet);
+
+        } catch (URISyntaxException ex) {
+            fail(ex.getMessage());
+            return null;
+        }
+    }
+
+    @Test
+    public void testTransparentTile() throws IOException, FactoryException {
+
+        Layer layer = new AsyncTileLayer(service);
+//        Layer layer = new TileLayer(service);
+
+        MapContent map = new SingleLayerMapContent(layer);
+
+        GTRenderer renderer = new StreamingRenderer();
+        renderer.setMapContent(map);
+
+        ReferencedEnvelope mapBounds = service.getBounds();
+        double heightToWidth = mapBounds.getSpan(1) / mapBounds.getSpan(0);
+        int imageWidth = 600;
+        Rectangle imageBounds = new Rectangle(0, 0, imageWidth, (int) Math.round(imageWidth * heightToWidth));
+        map.getViewport().setScreenArea(imageBounds);
+        map.getViewport().setBounds(mapBounds);
+
+        BufferedImage image = new BufferedImage(imageBounds.width, imageBounds.height, BufferedImage.TYPE_INT_RGB);
+        Graphics2D gr = image.createGraphics();
+
+        // set background color for assertion
+        gr.setPaint(Color.RED);
+        gr.fill(imageBounds);
+
+        // render tiles
+        renderer.paint(gr, imageBounds, mapBounds);
+//        File fileToSave = new File("d:\\test.bmp");
+//        ImageIO.write(image, "bmp", fileToSave);
+
+        // get first pixel
+        int argb = image.getRGB(0, 0);
+        boolean hasAlphaChannel = image.getAlphaRaster() != null;
+        Color c = new Color(argb, hasAlphaChannel);
+        System.out.println(String.format("First pixel : alpha=%s red=%s green=%s blue=%s", c.getAlpha(), c.getRed(), c.getGreen(), c.getBlue()));
```

Comment:
Please no `System.out` in tests. You have three. Use a logger so that builds can control their output. We have builds that fail because the log files are too large.

Reply:
removed, only needed for debug